### PR TITLE
fluidlite: added receipe for latest github version (1.0.9)

### PR DESCRIPTION
**Pull request by Freaxed**
Latest version of the fluidlite library. Among many fixes and improvements there a fix for the function 'fluid_synth_sfload' (return of -1 in case of missing or invalid file). The buggy function is also used in haiku, midi-kit, SoftSynth.cpp.

### CODE REVIEW

File: media-sound/fluidlite/fluidlite-1.0.9.recipe [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+srcGitRev="e64b4b3161cab212fffe7d1d3fb1c05750c363cc"
+SOURCE_URI="$HOMEPAGE/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="b6c4bff689bfd2b8e57b771640da019ee6cf3580a96ddced3954f9f84b66410f"
+SOURCE_DIR="FluidLite-$srcGitRev"
+
+ARCHITECTURES="all"
+SECONDARY_ARCHITECTURES="x86_gcc2 x86"
+
+PROVIDES="
+	fluidlite${secondaryArchSuffix} = $portVersion
```
Comment (Begasus) [inline media-sound/fluidlite/fluidlite-1.0.9.recipe:27]: Missing, and no brackets needed here:
```diff
- 	fluidlite${secondaryArchSuffix} = $portVersion
+ 	fluidlite$secondaryArchSuffix = $portVersion
+ 	lib:libfluidlite$secondaryArchSuffix = $portVersion
```

File: media-sound/fluidlite/fluidlite-1.0.9.recipe [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	"
+REQUIRES="
+	haiku${secondaryArchSuffix}
```
Comment (Begasus) [inline media-sound/fluidlite/fluidlite-1.0.9.recipe:30]: Missing:
```diff
+ 	lib:liboggsecondaryArchSuffix
+ 	lib:libvorbis$secondaryArchSuffix
+ 	lib:libvorbisfile$secondaryArchSuffix
```

File: media-sound/fluidlite/fluidlite-1.0.9.recipe [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	"
+
+PROVIDES_devel="
+	fluidlite${secondaryArchSuffix}_devel = $portVersion
+	devel:libfluidlite$secondaryArchSuffix = $portVersion compat >= 1
+	"
+REQUIRES_devel="
```
Comment (Begasus) [inline media-sound/fluidlite/fluidlite-1.0.9.recipe:37]: Missing:
```diff
+ 	fluidlite$secondaryArchSuffix == $portVersion base
+ 	devel:liboggsecondaryArchSuffix
+ 	devel:libvorbis$secondaryArchSuffix
+ 	devel:libvorbisfile$secondaryArchSuffix
```

File: media-sound/fluidlite/fluidlite-1.0.9.recipe [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libogg$secondaryArchSuffix
+	devel:libvorbis$secondaryArchSuffix
```
Comment (Begasus) [inline media-sound/fluidlite/fluidlite-1.0.9.recipe:43]: Missing:
```diff
+ 	devel:libvorbisfile$secondaryArchSuffix
```

File: media-sound/fluidlite/fluidlite-1.0.9.recipe [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX:PATH=$prefix .
```
Comment (Begasus) [inline media-sound/fluidlite/fluidlite-1.0.9.recipe:55]: Changes, enable SF3, use system vorbis/ogg:
```diff
- 	cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX:PATH=$prefix .
+ 	cmake -DCMAKE_BUILD_TYPE=Release \
+ 	 	$cmakeDirArgs \
+ 	 	-DENABLE_SF3=YES \
+ 	 	-DSTB_VORBIS=NO .
```

File: media-sound/fluidlite/fluidlite-1.0.9.recipe [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+INSTALL()
+{
+	make install
+
+	mv $prefix/lib $prefix/lib2
+	mkdir -p $(dirname $includeDir) $(dirname $libDir)
+	mv $prefix/include $includeDir
+	mv $prefix/lib2 $libDir
```
Comment (Begasus) [inline media-sound/fluidlite/fluidlite-1.0.9.recipe:66]: Not needed (cmake uses GNUInstallDirs)
```diff
-
-	mv $prefix/lib $prefix/lib2
-	mkdir -p $(dirname $includeDir) $(dirname $libDir)
-	mv $prefix/include $includeDir
-	mv $prefix/lib2 $libDir
```

File: media-sound/fluidlite/fluidlite-1.0.9.recipe [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+	prepareInstalledDevelLib libfluidlite
```
Comment (Begasus) [inline media-sound/fluidlite/fluidlite-1.0.9.recipe:68]: Move fluidlite.pc to devel package:
```diff
+ 	fixPkgconfig
```

**Review by Begasus — DISMISSED** [review]
Some changes done from previous version, shared library is build default, changes to make the build using system libraries ...

File: media-sound/fluidlite/fluidlite-1.0.9.recipe [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+fluid_settings_setstr(settings, "synth.drums-channel.active", "no"); \
+you can still select bank 128 on any channel to use drum kits.
+
+FluidLite keeps very minimal functionnalities (settings and synth), \
+therefore MIDI file reading, realtime MIDI events and audio output must be \
+implemented externally."
+HOMEPAGE="https://github.com/divideconcept/FluidLite"
+COPYRIGHT="2016 Robin Lobel"
+LICENSE="GNU LGPL v2.1"
+REVISION="2"
```
Comment (korli) [inline media-sound/fluidlite/fluidlite-1.0.9.recipe:17]:
```suggestion
REVISION="1"
```

**Comment by korli** [comment]
ideally removing the old recipe would help to see the differences applied.

**Comment by korli** [comment]
why is the version now 1.0.9 ? there isn't such upstream release or versioning that I could find

**Comment by Freaxed** [comment]
> why is the version now 1.0.9 ? there isn't such upstream release or versioning that I could find

I found the version in fluid_config.h.. I agree, there is no official release on github but because my original idea was to keep the old version I searched for something in the code.. BTW, also version 1.0.0 is quite unofficial..

**Comment by korli** [comment]
Fine, thanks for the answer.

**Comment by Begasus** [comment]
Thanks! :+1:

**Pull request merged**